/* $DragonFly: src/usr.bin/vi/patches/v_ex.c.patch,v 1.1 2006/02/26 17:18:06 dillon Exp $ */
--- v_ex.c	2003-06-17 10:46:12.000000000 +0800
+++ v_ex.c	2006-02-26 20:05:01.000000000 +0800
@@ -198,11 +198,16 @@
 	 * Try the alternate file name, then the previous file
 	 * name.  Use the real name, not the user's current name.
 	 */
-	if ((name = sp->alt_name) == NULL) {
+	if (sp->alt_name == NULL) {
 		msgq(sp, M_ERR, "180|No previous file to edit");
 		return (1);
 	}
 
+	if ((name = strdup(sp->alt_name)) == NULL) {
+		msgq(sp, M_SYSERR, NULL);
+		return (1);
+	}
+
 	/* If autowrite is set, write out the file. */
 	if (file_m1(sp, 0, FS_ALL))
 		return (1);
